build.rs

```diff
@@ -1,6 +1,30 @@
 use std::env;
 
+include!("src/dist/triple.rs");
+
+pub fn from_build() -> Result<PartialTargetTriple, String> {
+    if let Some(triple) = env::var("RUSTUP_OVERRIDE_BUILD_TRIPLE").ok() {
+        PartialTargetTriple::new(&triple).ok_or(triple)
+    } else {
+        let triple = env::var("TARGET").unwrap();
+        PartialTargetTriple::new(&triple).ok_or(triple)
+    }
+}
+
 fn main() {
+    println!("cargo:rerun-if-env-changed=RUSTUP_OVERRIDE_BUILD_TRIPLE");
+    println!("cargo:rerun-if-env-changed=TARGET");
```

Comment:
I think you might also need to tell cargo if the build script changes? Not sure if it infers that automatically.
```suggestion
    println!("cargo:rerun-if-env-changed=TARGET");
    println!("cargo:rerun-if-changed=build.rs");
```

Reply:
Cargo always tracks the *source* inputs to the build script automatically.